f8a_report/sentry_report_helper.py

```diff
@@ -0,0 +1,132 @@
+            if not result['error_report'][server_name].get('errors'):
+                result['error_report'][server_name]['errors'] = []
+            result['error_report'][server_name]['errors'].append(errors)
+            # Saving the final report in the relevant S3 bucket
+
+            try:
+                obj_key = '{depl_prefix}/{type}/{report_name}.json'.format(
+                    depl_prefix=self.s3.deployment_prefix, type=report_type, report_name=report_name
+                )
+                self.s3.store_json_content(content=result, obj_key=obj_key,
+                                           bucket_name=self.s3.report_bucket_name)
+            except Exception as e:
+                logger.exception('Unable to store the report on S3. Reason: %r' % e)
+
+        return result
+
+    def retrieve_events(self, issue_id):
+        """Retrieve results for issue events."""
+        events = {'stacktrace': ''}
+
+        try:
+            # Invoke Sentry API to run the event collection
+            auth = 'Bearer {token}'.format(token=self.sentry_token)
+            resp = requests.get(url=self.sentry_api_tags + issue_id + '/events/latest/',
+                                headers={"Authorization": auth})
+            # Check for status code
+            # If status is not success, log it as an error
+            if resp.status_code == 200:
+                logger.info('Successfully invoked Sentry API \n {resp}'.format(resp=resp.json()))
+                output = resp.json()
+                # retrieving server name info
+                for item in output['tags']:
+                    if item['key'] == 'server_name':
+                        events['pods_impacted'] = item['value']
+                        break
+                # Collecting stacktrace for each frames
+                if output['entries'][1]['type'] == 'exception':
+                    for frames in output['entries'][1]['data']['values'][0]['stacktrace']['frames']:
+                        stacktrace = 'File ' + frames['filename'] + ', Line ' + \
+                                     str(frames['lineNo']) + ', Function ' + \
+                                     frames['function']
+                        # Collecting stacktrace for each line Nos
+                        for context in frames['context']:
+                            if frames['lineNo'] in context:
+                                stacktrace = stacktrace + ', Statement ' + \
+                                             context[1] + ' || '
+                                break
+                        else:
+                            stacktrace = stacktrace + ' || '
+                        events['stacktrace'] = events['stacktrace'] + stacktrace
+                else:
+                    events['stacktrace'] = 'Not Available'
+            else:
+                logger.error('Error received from Sentry API \n {resp}'.format(resp=resp.json()))
+        except Exception as e:
```

Comment:
These are all broad exceptions. Please provide more detailed exceptions level

Reply:
changed accordingly. thanks for suggestion